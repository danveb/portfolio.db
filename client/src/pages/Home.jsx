import chevronDown from "../assets/hero/chevron-down.svg"; 
import Button from "../components/Button";
import "../styles/Home.css"; 

export default function Home() {
  return (
    <div className="home">
      <div className="home__wrapper">
        <section className="home__top">
          <div className="home__title">
            <h1>hey, I'm Danny<span className="dot--green">.</span> üë®‚Äçüíª</h1>
          </div>
          <div className="home__intro">
            <p>I'm a frontend developer, passionate about finding ways to efficiently solve complex problems and make a positive difference for today and for tomorrow. I'm currently building my personal portfolio, which contains some of my latest projects. </p>
          </div>
          <div className="home__intro">
            <p>Feel free to continue browsing my portfolio to know a little more about me, and the type of projects I love to create. </p>
          </div>
          <div className="home__intro">
            <p>Over the past couple of years, I've dived into the technologies I'm interested in at the time, and how I learned to grow in the development world. </p>
          </div>
          <div className="home__intro">
            <p><span className="dot--green bold">Happy coding!</span> </p>
          </div>
          {/* <div className="hero__arrow">
            <img src={chevronDown} alt="chevron down arrow" />
          </div> */}
          <Button title={"Explore Projects"} url="/projects" />
        </section>
      </div>
    </div>
  )
}