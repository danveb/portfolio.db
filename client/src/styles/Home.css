.home {
  width: 100%; 
  height: 100%; 
}

.home__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
  width: 100%; 
  height: 100%; 
  background-color: var(--rich-black); 
}

.home__top {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 40px 0;
}

.home__title {
  font-family: "Mulish", sans-serif; 
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.home__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
}

@media screen and (max-width: 1024px) {
  .home__wrapper {
    padding: 0 20px; 
  }
}

@media screen and (max-width: 768px) {
  .home__title {
    font-size: 16px; 
  }

  .home__intro {
    font-size: 14px; 
  }
}