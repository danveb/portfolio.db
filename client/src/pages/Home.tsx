import { Button, Gallery } from "../components"; 
import "../styles/Home.css"; 

export default function Home() {
  return (
    <div className="home">
      <div className="home__wrapper">
        <section className="home__top">
          <div className="home__title">
            <h1>hey, I'm Danny<span className="dot--green">.</span> üë®‚Äçüíª</h1>
          </div>
          <div className="breakline"></div>
          <div className="home__intro">
            <p>I'm a <span>frontend developer</span>, passionate about finding ways to efficiently solve complex problems and make a positive difference for today and for tomorrow. I'm currently building my personal portfolio, which contains some of my latest projects. </p>
          </div>
          <div className="home__intro">
            <p>Over the past couple of years, I've dived into the technologies I'm interested in at the time, and how I learned to grow in the development world. </p>
          </div>
          <div className="home__intro">
            <p>Feel free to continue browsing my portfolio to know a little more about me, and the type of projects I love to create. </p>
          </div>
          <Button title="read my story" url="/about" />
        </section>
        <section className="home__bottom">
          <Gallery />
          <Button title="explore projects" url="/projects" />
        </section>
      </div>
    </div>
  )
}