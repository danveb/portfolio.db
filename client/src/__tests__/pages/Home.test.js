import { render, screen } from "@testing-library/react";
import Home from "../../pages/Home";
import { BrowserRouter } from "react-router-dom";

describe("Home Component", () => {
    // smoke test
    test("renders without crashing", () => {
      render(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      );
    });

    // snapshot test 
    test("matches snapshot", () => {
      const { asFragment } = render(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      );
      expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText h1
    test("heading tag is displayed correctly", () => {
      render(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      );
      const heading1 = screen.getByText("hey, I'm Danny", {
        exact: false
      });
      expect(heading1).toBeInTheDocument(); 
    });
}); 