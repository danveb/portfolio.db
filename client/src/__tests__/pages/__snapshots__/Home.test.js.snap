// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="home__wrapper"
    >
      <section
        class="home__top"
      >
        <div
          class="home__title"
        >
          <h1>
            hey, I'm Danny
            <span
              class="dot--green"
            >
              .
            </span>
             üë®‚Äçüíª
          </h1>
        </div>
        <div
          class="breakline"
        />
        <div
          class="home__intro"
        >
          <p>
            I'm a 
            <span>
              frontend developer
            </span>
            , passionate about finding ways to efficiently solve complex problems and make a positive difference for today and for tomorrow. I'm currently building my personal portfolio, which contains some of my latest projects. 
          </p>
        </div>
        <div
          class="home__intro"
        >
          <p>
            Over the past couple of years, I've dived into the technologies I'm interested in at the time, and how I learned to grow in the development world. 
          </p>
        </div>
        <div
          class="home__intro"
        >
          <p>
            Feel free to continue browsing my portfolio to know a little more about me, and the type of projects I love to create. 
          </p>
        </div>
        <div
          class="button__container"
        >
          <a
            class="button--cta"
            href="/about"
          >
            read my story
            <img
              alt="right arrow"
              src="right-arrow.png"
            />
          </a>
        </div>
      </section>
      <section
        class="home__bottom"
      >
        <div
          class="gallery"
        >
          <div
            class="gallery__wrapper"
          >
            <div
              class="gallery__card"
            >
              <img
                alt=""
                src="sailly_blank.png"
              />
              <img
                alt=""
                src="asador_blank.png"
              />
              <img
                alt=""
                src="bobaLife_blank.png"
              />
              <img
                alt=""
                src="ioniq5_blank.png"
              />
              <img
                alt=""
                src="coini$t_blank.png"
              />
            </div>
          </div>
        </div>
        <div
          class="button__container"
        >
          <a
            class="button--cta"
            href="/projects"
          >
            explore projects
            <img
              alt="right arrow"
              src="right-arrow.png"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
